management-server/internal/xds/server.go

```diff
@@ -81,16 +85,175 @@ func FeedData() {
 	newSnapshot, _ := wso2_cache.NewSnapshot(fmt.Sprint(version), map[wso2_resource.Type][]types.Resource{
 		wso2_resource.APKMgtApplicationType: {&applications},
 	})
-	apiCacheMutex.Lock()
-	// TODO (amaliMatharaarachchi) update relevant adapters with snapshot caches
-	apiCache.SetSnapshot(context.Background(), config.ManagementServer.NodeLabels[0], newSnapshot)
-	apiCacheMutex.Unlock()
+	applicationCacheMutex.Lock()
+	defer applicationCacheMutex.Unlock()
+	applicationCache.SetSnapshot(context.Background(), "mine", newSnapshot)
```

Comment:
```suggestion
	applicationCache.SetSnapshot(context.Background(), config.ManagementServer.NodeLabels[0], newSnapshot)
```

Reply:
fixed in https://github.com/wso2/apk/pull/151/commits/4f4c37b97abe8646e783d357a3998601be6de56c